sstables/sstables_manager.cc

```diff
@@ -334,6 +334,8 @@ std::vector<std::filesystem::path> sstables_manager::get_local_directories(const
 }
 
 void sstables_manager::on_unlink(sstable* sst) {
+    // Reclaim memory used by the sstable components
+    reclaim_memory_from_sstable(*sst);
```

Comment:
How does this prevent the sstable from being selected for reload?

Reply:
The sstable with the highest memory usage is reclaimed whenever a new sstable is added, and the total component memory, including the new sstable's memory, exceeds the threshold. This guarantees that whenever a reclamation code is triggered, there exists atleast one sstable whose component memory is non zero, ensuring that any sstable with components memory == 0 will never be picked up for reclamation.

Relevent code here - https://github.com/scylladb/scylladb/blob/98860905d820efa15458458e9c93f148fd81b62f/sstables/sstables_manager.cc#L159-L168

Reply:
Shall I add the following line after `auto sst_with_max_memory = std::max_element...`?
```
    if (sst_with_max_memory->total_reclaimable_memory_size() == 0) {
        return;
    }
```

As per the logic I explained above this `if` will never be true but it will make the code more clear.

Reply:
I ask about the other case: when we have some free memory and want to reload some previously evicted sstable component. How will the above prevent an unlinked sstable from being picked for reloading its component?

Reply:
Only sstables from `_reclaimed` set will be picked up for reload. Unlinked table will never end up in that set for two reasons 
1. As explained above they will never be picked up for reclaim and hence never be reloaded
2. If they were already in `_reclaimed` when they are unlinked, we explicitly remove them from that set. (#19722)

Reply:
I see, I though the bug we fix here is that we unlink sstables that are already reclaimed, but after re-reading the issue, I realized the bug is that we reclaim already unlinked sstables (then reload them later).